packages/suite/src/components/suite/CountdownTimer.tsx

```diff
@@ -1,46 +1,73 @@
-import React, { useEffect, useState } from 'react';
-import { useLocales } from '@suite-hooks/useLocales';
-import { formatTimeLeft } from '@suite-utils/formatTimeLeft';
+import React from 'react';
+import styled from 'styled-components';
+import { FormattedNumber } from 'react-intl';
 import { Translation } from './Translation';
+import { TranslationKey } from '@suite-common/intl-types';
+import { useCountdownTimer } from '@suite-hooks';
 
-interface TimerProps {
+const Wrapper = styled.span`
+    font-variant-numeric: tabular-nums;
+`;
+
+type Unit = 'second' | 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year';
+
+const units: Array<{ format: keyof Duration; unit: Unit }> = [
+    { format: 'years', unit: 'year' },
+    { format: 'months', unit: 'month' },
+    { format: 'weeks', unit: 'week' },
+    { format: 'days', unit: 'day' },
+    { format: 'hours', unit: 'hour' },
+    { format: 'minutes', unit: 'minute' },
+    { format: 'seconds', unit: 'second' },
+];
+
+interface CountdownTimerProps {
     deadline: number | string;
-    format?: Array<keyof Duration>;
+    pastDeadlineMessage?: TranslationKey;
+    minUnit?: Unit;
+    minValue?: number;
+    unitDisplay?: 'long' | 'short' | 'narrow';
```

Comment:
nit: `NumberFormatOptionsUnitDisplay`

Reply:
I don't want to import @formatjs/ecma402-abstract for it...

Reply:
I'm sorry, I didn't notice it's not from `react-intl`... I was curios if there are solutions for situations like that, and `infer` [seems to be it](https://www.totaltypescript.com/tips/write-your-own-propsfrom-helper-to-extract-props-from-any-react-component) :) Just FYI, not asking to change anything.